client.go

```diff
@@ -171,7 +171,12 @@ func (c *Client) setRequestHeaders(req *http.Request, op *internal.Op) error {
 
 	switch op.Endpoint {
 	case internal.API:
-		req.SetBasicAuth(c.skey, "")
+		// Override to use public key for some api endpoint e.g. /sources
```

Comment:
Feels hacky 🤔  Can we do something like:

```go
switch selectKeyKind(op) {
case "public":
  req.setBasicAuth(c.pkey, "")
case "secret":
  req.setBasicAuth(c.skey, "")
}

func (c *Client) selectKeyKind(op *internal.Op) string {
  switch {
  case op.APIKey != "":
    return op.APIKey
  case op.Endpoint == internal.API:
    return "secret"
  case op.Endpoint == internal.Vault:
    return "public"
  }
}
```

instead krub? It's a bit more flexible this way and move the knowledge of which key to use back to operations object instead of *both* the operation object and the client.

Reply:
👍 I agree. And I think method for lookup `key_kind` should be method of `*internal.Op`.